File: guides/claude/claude-managed-agents/create_environment.py
"""Create a self-hosted environment on Claude Platform."""
from __future__ import annotations

import os

import anthropic
import dotenv

dotenv.load_dotenv(override=True)

client = anthropic.Anthropic(api_key=os.environ["ANTHROPIC_API_KEY"])

environment = client.beta.environments.create(
    name="superserve",
    config={"type": "self_hosted"},
)

print(f"created environment {environment.id}")
print()
print("next steps:")
print(f"  1. open https://platform.claude.com/workspaces/default/environments/{environment.id}")
print("  2. click 'Generate environment key'")
print("  3. add both values to .env:")
print(f"     ANTHROPIC_ENVIRONMENT_ID={environment.id}")
print("     ANTHROPIC_ENVIRONMENT_KEY=sk-ant-oat01-...")

File: guides/managed-agents/claude-managed-agents/typescript/create-environment.mjs
import "dotenv/config"
import Anthropic from "@anthropic-ai/sdk"

const client = new Anthropic()

const environment = await client.beta.environments.create({
  name: "superserve",
  config: { type: "self_hosted" },
})

console.log(`created environment ${environment.id}`)
console.log()
console.log("next steps:")
console.log(`  1. open https://platform.claude.com/workspaces/default/environments/${environment.id}`)
console.log("  2. click 'Generate environment key'")
console.log("  3. add both values to .env:")
console.log(`     ANTHROPIC_ENVIRONMENT_ID=${environment.id}`)
console.log("     ANTHROPIC_ENVIRONMENT_KEY=sk-ant-oat01-...")
